schema.development.kf

```diff
@@ -637,70 +618,107 @@ action share_credential_through_dag (
     );
 }
 
-action share_credential_by_write_grant (
-    $id,
-    $user_id,
+action create_credentials_by_dwg(
+    $issuer_auth_public_key,
     $encryptor_public_key,
-    $content,
-    $content_hash,
-    $public_notes,
-    $public_notes_signature,
-    $broader_signature,
     $original_credential_id,
-    $grantee_wallet_identifier,
-    $locked_until,
-    $issuer_auth_public_key
-) public {
-    SELECT CASE WHEN $locked_until < 0 THEN
-        error('locked_until must be positive integer timestamp or zero')
-    END;
-
-    SELECT get_write_grant_id($user_id); // throw an error if no write_grant
+    $original_content,
+    $original_public_notes,
+    $original_public_notes_signature,
+    $original_broader_signature,
+    $copy_credential_id,
+    $copy_content,
+    $copy_public_notes_signature, // Signature of an empty string?! This is wierd.
+    $copy_broader_signature,
+    $dwg_owner,
+    $dwg_id,
+    $dwg_not_before,
+    $dwg_not_after,
+    $dwg_signature) public {
 
-    SELECT CASE WHEN NOT credential_belongs_to_user($original_credential_id, $user_id) THEN
-        error('the original credential does not belong to the user')
+    SELECT CASE
+        WHEN NOT EXISTS (SELECT 1 FROM wallets WHERE (wallet_type = 'EVM' AND address=$dwg_owner COLLATE NOCASE)
+            OR (wallet_type = 'NEAR' AND public_key = $dwg_owner))
+        THEN ERROR('dwg_owner not found')
     END;
 
-    SELECT CASE WHEN get_write_grant_id($user_id)::text != get_credential_inserter($original_credential_id) THEN
-        error('you can share only original credentials you created')
-    END;
+    $times_validation = idos.validate_not_usable_times($dwg_not_before, $dwg_not_after); // Check the format and precedence
+    SELECT CASE WHEN $times_validation != 1 THEN ERROR('dwg_not_before must be before dwg_not_after') END;
 
-    SELECT CASE WHEN (SELECT count(1) FROM access_grants WHERE id = uuid_generate_v5('31276fd4-105f-4ff7-9f64-644942c14b79'::uuid, format('%s-%s-%s', $grantee_wallet_identifier::text, $id::text, $locked_until::text))) > 0 THEN
-        error('a grant with the same grantee, original_credential_id, and locked_until already exists')
-    END;
+    $dwg_result = idos.dwg_veryfy_owner($dwg_owner, @caller, $dwg_id, $dwg_not_before, $dwg_not_after, $dwg_signature);
```

Comment:
I would not assume that the caller is the grantee. I'd say that anybody can transmit this transaction (to enable flows where the IdV is the one integrating idOS, like in Fractal's case).

Reply:
Hmm, you are right. I will fix it.

Reply:
Changed by this https://github.com/idos-network/idos-schema/pull/68/commits/6d717ddbfa20ba7a588a76233a37439e47b46e01